### PR TITLE
STAR-239 adding canGrow to columndefinition

### PR DESCRIPTION
**Pull request by oscarkwan**
### Purpose 🚀
- Adding isGrow prop to ColumnDefinition to allow column to expand the available space.

### Notes ✏️
- Fixed typo with `useColumnsArrangement`
- Using `max-width` instead of `width`

### Updates 📦
- [x] MAJOR (breaking) change to _ActionBar_
- [x] MINOR (backward compatible) change to _DataGrid_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/STAR-239-column-resize-datgrid

### CODE REVIEW

File: packages/DataGrid/src/DataGrid.js [hunk 1]
```diff
@@ -98,6 +100,9 @@ const DataGrid = React.forwardRef((props, ref) => {
     start: null,
     stop: null,
   });
+  const totalColumnWidth = React.useRef(0);
```
Comment (nahumzs) [inline packages/DataGrid/src/DataGrid.js:103]: please add prefix `ref`  or would be hard to know that `.current` is required

**Comment by jamiek-acl** [comment]
@oscarkwan how about addressing this issue here:

- imagine you have a 3 column DataGrid (`name`, `gender`, `age`) with the widths 80%, 10%, 10% respectively.

    <DataGrid.ColumnDefinition header="name" width="800" />
    <DataGrid.ColumnDefinition header="gender" width="100" />
    <DataGrid.ColumnDefinition header="age" width="100" />

- then using ActionBar i re-arrange the columns so the order is now: `gender`, `name`, `age`.
- now `gender` is 80% width and `name` is only 10%.

The 80% width should be sticking with the `name` column.

Nahum mentioned it is happening `because its cached, the width of column is cached and doesn't get recalculated on the rerender`

**Comment by jamiek-acl** [comment]
`isGrow` is a weird name... `canGrow`?

File: packages/DataGrid/stories/DataGrid.resize.stories.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            <ColumnsArrangement.ColumnDefinition
+              id="country"
+              label="Country"
+              isDisabled={false}
+              isHidden={isColumnHidden("country")}
+            />
+            <ColumnsArrangement.ColumnDefinition
+              id="joined"
+              label="Joined"
+              isDisabled={false}
```
Comment (jamiek-acl) [inline packages/DataGrid/stories/DataGrid.resize.stories.js:126]: All of these `isDisabled={false}` are redundant, it is false by default

File: packages/DataGrid/stories/DataGrid.resize.stories.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+              isHidden={isColumnHidden("joined")}
+            />
+          </ColumnsArrangement>
+        </ActionBar>
+        <DataGrid key={orderedColumnIds.join("-")} data={data} width={1000}>
+          {orderedColumnIds.map(columnKey =>
+            isColumnHidden(columnKey) ? null : (
```
Comment (jamiek-acl) [inline packages/DataGrid/stories/DataGrid.resize.stories.js:133]: Suggestion, you dont have to take it:

`!isColumnHidden(columnKey) && (<DataGrid.ColumnDefinition ...`

File: packages/DataGrid/src/DataGrid.js [hunk 4]
```diff
@@ -98,6 +100,9 @@ const DataGrid = React.forwardRef((props, ref) => {
     start: null,
     stop: null,
   });
+  const reftotalColumnWidth = React.useRef(0);
```
Comment (jamiek-acl) [inline packages/DataGrid/src/DataGrid.js:103]: capitalization

File: packages/DataGrid/src/DataGrid.js [hunk 5]
```diff
@@ -436,6 +442,27 @@ const DataGrid = React.forwardRef((props, ref) => {
     });
   }, [gridId, data]);
 
+  React.useEffect(() => {
+    if (!refContainer.current) return;
+
+    if (refGridColumns.current !== null && refGridStickyHeader.current !== null && refGridHeader.current !== null) {
```
Comment (jamiek-acl) [inline packages/DataGrid/src/DataGrid.js:448]: How about
`if (refGridColumns.current && refGridStickyHeader.current && refGridHeader.current)`?

File: packages/DataGrid/src/DataGrid.js [hunk 6]
```diff
@@ -436,6 +442,27 @@ const DataGrid = React.forwardRef((props, ref) => {
...
+      refGridHeader.current.resetAfterColumnIndex(0);
+      refGridStickyHeader.current.resetAfterColumnIndex(0);
+      refGridColumns.current.resetAfterColumnIndex(0);
+    }
+
+    ColumnDefinitions.forEach(columnDefinition => {
+      reftotalColumnWidth.current = columnDefinition.props.width + reftotalColumnWidth.current;
```
Comment (jamiek-acl) [inline packages/DataGrid/src/DataGrid.js:455]: `reftotalColumnWidth.current += columnDefinition.props.width;`?

**Pull request merged**